### ISSUE TITLE
Discuss how to support shuttle build events

### ISSUE THREAD

**Issue by optimumtact** [tgstation/-tg-station#6856]
These are events admins like to run where the crew is forced to build a new shuttle.

With the recent change to the shuttle controller it's unclear if this is still possible. Rumour has it that the new shuttle controller will only move shuttle tiles

Can we 
A) clarify what type of tiles will be moved by the new controller
B) Discuss what is needed to resupport this popular event type

Do we perhaps need a special dock type that moves all items within it? I think with the new modular dock system we can make a subchild that would do that.

**Comment by optimumtact** [comment]
@Carn can you please answer A) when you get a chance? :smiley_cat:

**Comment by optimumtact** [comment]
tkdrg suggested we make shuttle tiles buildable so the crew can build those instead, thats another good option

**Comment by Stickymayhem** [comment]
An acceptable solution but it does limit the crews ability to make a customised shuttle as the shape is strictly defined and so much of the actual design comes from materials used.

**Comment by tkdrg** [comment]
Shape isn't strictly defined now. Admins can easily edit that.

**Comment by optimumtact** [comment]
Yeah I think mins could edit a docking ports values to modify the bounding box as well

**Comment by Carn** [comment]
A) It will only move turfs of type /turf/simulated/shuttle

B)
The long-run plan was to make shuttle turfs buildable only within stationary docking port bounds.
It's more important that I get rotation and such working first (which means work on some form of smoothwall or rotation of shuttle turfs, as well as wall-mounted computers and such. e.g. shuttle timer displays)

It wasn't my intention to impede these events, I've never seen one held. It was a change merely to allow the new system to differentiate between turfs it wants to move and turfs from the surrounding area. For an example of why it was necessary, look at the syndicate shuttle. Without this change it would have moved all those snow turfs to another location. Moreover, a lot of surrounding turfs are unsimulated, which could cause serious problems if they're next to simulated ones (unless this was an issue that has been resolved in the past 2-3 years?). Either way it'd look silly.

Another thing I want to do after that is make shuttle engines buildable. More engines == shorter hyperspace transit sequence. No engines == no move. etc. However, how would this affect the escape shuttle?

tl;dr: I need to work out a few usability and design issues with shuttles, then I'll make shuttle turfs buildable. But I have other things I need to do too, so this issue may take a while.

I dunno if you guys want to use this issue as a brainstorming session or whatever, just keep it feasible! So that this doesn't become a pipe dream.

Thanks

**Comment by Carn** [comment]
Also yes, mins can edit bounding boxes.

An idea somebody suggested is have some sort of system where normal turfs are converted into shuttle turfs when certain conditions are met. I'm not sure how one would implement this though. I wouldn't want to have to use some obscure new item to make a shuttle, that's a lazy design habit that occurs way too much in ss13.

**Comment by Carn** [comment]
Also, I'm not sure if we should think about standardising shuttles a little bit into size-classes. This would make map-editting much easier, and make shuttles compatible with more stationary docking ports. At the moment the sizes vary a lot, and you can only dock a shuttle in a port if the shuttle's boundingbox fits inside the stationary-port's bounding box. So things like the shuttle and syndicate ship aren't compatible with any ports but their own.

**Comment by Jalleo** [comment]
It was talked about before with standardising shuttles but it was thrown out for a silly reason I believe if classing them would make them more expandable and more easily defineable then I would say we should do that.

**Comment by Carn** [comment]
Well, I can do that fairly easily.
I'll do it tomorrow night

**Comment by Dorsisdwarf** [comment]
Abandoned discussion thread with someone who has ceased to code fo tgstation, requesting closure by @Cheridan

**Comment by RemieRichards** [comment]
Area based shuttles used to move everything in their contents, so why do these new ones lock it to /turf/simulated/shuttle subtypes? I'll have a mess around tonight but I think it may be fine to just remove that type check.

**Comment by Jalleo** [comment]
I think there was probably a reason carn mentioned but I probably wasnt there when it was mentioned if anyone knows it they should mention it now

**Comment by optimumtact** [comment]
They move everything inside the shuttle zone, so I believe carn was having issues with half the syndicate station coming along for the ride.
